autoscheduler/frontend/src/components/NavBar.tsx

```diff
@@ -44,15 +90,20 @@ const NavBar: React.SFC = () => {
               </Typography>
             </Button>
           </div>
-
-          <Button
-            color="inherit"
-            onClick={(): void => {
-              window.open('/login/google-oauth2/', '_self');
-            }}
-          >
-            Google Login
-          </Button>
+          {usersName === ''
+            ? (
+              <div>
+                {LoginButton()}
+              </div>
+            )
+            : (
+              <div>
+                <Typography variant="subtitle1">
+                  {usersName/* make font prettier */}
+                  {LogoutButton()}
+                </Typography>
+              </div>
+            )}
```

Comment:
I like to avoid putting as little logic as possible into the component that gets rendered, so that you tell at a glance how the components are nested rather than looking at the business logic. You can do a couple things to help this:
- Move the ternary logic outside of the JSX
- Make the login/logout buttons include the typography, username, etc. instead of just the `Button` component

and for this part of the code, just put the variable that has the correct components to render:
```suggestion
          {userInformation}
```